aws/data_source_aws_service_discovery_dns_namespace.go

```diff
@@ -0,0 +1,100 @@
+package aws
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/servicediscovery"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+)
+
+func dataSourceServiceDiscoveryDnsNamespace() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceServiceDiscoveryDnsNamespaceRead,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"dns_type": {
```

Comment:
I suggest renaming this to simply `type` to match the AWS API.

Reply:
@ewbankkit  Done